packages/flutter_tools/lib/src/commands/build_bundle.dart

```diff
@@ -108,6 +109,11 @@ class BuildBundleCommand extends BuildSubCommand {
   @override
   Future<FlutterCommandResult> runCommand() async {
     final String targetPlatform = stringArg('target-platform')!;
+    final String channel = globals.flutterVersion.channel;
+    final bool onStableOrBetaChannel = channel == 'stable' || channel == 'beta';
+    if ((targetPlatform == 'windows-arm64') && onStableOrBetaChannel) {
+      throwToolExit('Option "--target-platform windows-arm64" can only be used on master channel.');
+    }
```

Comment:
For now, we want to disallow cross-compilation on Windows. Should this instead fail if `--target-platform windows-arm64` is used on a Windows x64 host, or, if `--target-platform windows-x64`  current platform ABI does not match the target platform?

If yes, please add a test to cover this scenario.

Reply:
You're right, and after our discussion, we decided to remove this option. I removed it from build command but forgot to modify this one as well.
I removed this part and updated associate build_bundle test.